File: HealthCare/HealthCare.Shared/ApiResponse.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace HealthCare.Shared
{
    public class ApiResponse<T>
    {
        public T? Data { get; set; }
        public bool Success { get; set; }
        public string? Message { get; set; }
        public List<string>? Errors { get; set; }
    }
} 

File: HealthCare.Portal/src/app/core/models/api-response.ts
export interface ApiResponse<T> {
  success: boolean;
  message: string;
  data: T;
  errors: string[] | null;
}

export interface PagedResult<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}

export interface PaginationParams {
  pageNumber?: number;
  pageSize?: number;
}

export interface ErrorResponse {
  statusCode: number;
  message: string;
  details: string | null;
  timestamp: string;
  path: string;
}
